HabitRPG/HRPGModels/User.m

```diff
@@ -157,7 +161,21 @@ - (void)getAvatarImage:(void (^)(UIImage *))successBlock withPetMount:(BOOL)with
             dispatch_group_leave(group);
         }];
     }
+    
+    NSString *back = [self.useCostume boolValue] ? self.costumeBack : self.equippedBack;
+    if (![back isEqualToString:@"back_base_0"]) {
+        NSString *format = nil;
+        dispatch_group_enter(group);
+        [sharedManager getImage:[NSString stringWithFormat:@"%@", back] withFormat:format onSuccess:^(UIImage *image) {
+            [imageArray replaceObjectAtIndex:currentLayer withObject:image];
+            dispatch_group_leave(group);
+        } onError:^() {
+            dispatch_group_leave(group);
+        }];
+    }
+    
     dispatch_group_enter(group);
+    currentLayer++;
```

Comment:
This should be inside the if statement above it. Otherwise the skin would always be at position 1 and position 0 could be empty, if no back gear is equipped